Core/Src/CAN/CAN_IDs.c

```diff
@@ -1,99 +1,94 @@
-/*
- * CAN_IDs.c
- *
- *  Created on: Dec 9, 2020
- *      Author: renanmoreira
- */
 #include "CAN/CAN_IDs.h"
 
-CAN_ID_t CAN_ID_map[CAN_ID_QUAN];
-// TODO(renanmoreira): talvez aumentar capacidade se precisar de mais ids de debug
-uint16_t
-    CAN_ID_POS_INTERNAL_ID_MAP[(ECU_CAN_LAST_POPULATED_ID - ECU_CAN_INITIAL_ID + 1) * 4
-                               + 8];
+#include <stdio.h>
+#include <string.h>
 
-uint16_t convert_id_pos_to_index(uint16_t id, uint16_t pos) {
-    return (id - ECU_CAN_INITIAL_ID) * 4 + pos;
+static CAN_var_inf CAN_ID_map[CAN_GENERAL_ID_QUAN];
+
+static datalog_send_t datalog_send_struct[QUANT_RESERVED_ID];
+
+static datalog_send_t aux_datalog_struct;
+
+static uint16_t quant_of_ext_id;
+
+static void populate_datalog_send_struct();
+static void initialize_CAN_var_inf();
+static int compare_function();
+static void sort_struct();
+
+static void initialize_CAN_var_inf() {
+    // does the mapping of the ids in a structure that will be sorted
+#define CAN_GENERAL_LIST_DATA(var_name, msg_id, msg_wrd)                                 \
+    CAN_ID_map[var_name].var = var_name;                                                 \
+    CAN_ID_map[var_name].id  = msg_id;                                                   \
+    CAN_ID_map[var_name].pos = msg_wrd;
+    VARIABLES_GENERAL_CAN_TX
+#undef CAN_GENERAL_LIST_DATA
 }
 
-CAN_ID_t get_CAN_ID_from_internal(uint16_t internal) {
-    return CAN_ID_map[internal];
+static int compare_function(const void* value1, const void* value2) {
+    // Sort by ID
+    if (((CAN_var_inf*)value1)->id < ((CAN_var_inf*)value2)->id) {
+        return -1;
+    }
+    if (((CAN_var_inf*)value1)->id > ((CAN_var_inf*)value2)->id) {
+        return +1;
+    }
```

Comment:
Para ficar menos verboso:
```suggestion
    const uint16_t first_id  = ((CAN_var_inf*)value1)->id;
    const uint16_t second_id = ((CAN_var_inf*)value2)->id;
    if (first_id < second_id) {
        return -1;
    }
    if (first_id > second_id) {
        return 1;
    }
```
E msm coisa para o pos